src/gc/impl/conservative/gc.d

```diff
@@ -267,21 +273,53 @@ struct GC
 
     import core.internal.spinlock;
     static gcLock = shared(AlignedSpinLock)(SpinLock.Contention.lengthy);
-    static bool inFinalizer;
+    static bool _inFinalizer;
 
     // lock GC, throw InvalidMemoryOperationError on recursive locking during finalization
     static void lockNR() @nogc nothrow
     {
-        if (inFinalizer)
+
+        if (_inFinalizer)
+        {
             onInvalidMemoryOperationError();
+        }
         gcLock.lock();
     }
 
-    __gshared Config config;
 
-    void initialize()
+    static void initialize()
     {
-        config.initialize();
+        import core.stdc.string: memcpy;
+
+        if(config.gc != "conservative")
+              return;
+
+        auto p = cstdlib.malloc(__traits(classInstanceSize,ConservativeGC));
+
+        if(!p)
+            onOutOfMemoryErrorNoGC();
+
+        instance = cast(ConservativeGC)memcpy(p, typeid(ConservativeGC).initializer.ptr, typeid(ConservativeGC).initializer.length);
+
+        instance.__ctor();
+
+        gc_setProxy(instance);
```

Comment:
This call is causing the test failure on the windows build server: it causes the exported function to be included in the binary, which makes the linker produce an import library. I'd rather return the instance and assign it to the global. (I still don't like checking `config.gc` in each GC, but that could be a matter of taste).